### DISCUSSION TITLE
How to include waves?

### DISCUSSION THREAD

**Discussion by CyprienBosserelle** [CyprienBosserelle/BG_Flood#76] [Ideas]
## What should be the priority for BG_Flood to include the impact of waves?

1. XBGPU style group resolving
2. Multi-layer non-hydrostatic solver (Basilisk)
3. Boussinesq  (Serre-Green-Naghdi) (Basilisk)
4. Other wave model

**Comment by ebeetham** [comment]
This would be so useful! I think 1) XBGPU style would be most useful for running multi-hazard models with a coastal storm and catchment flow. Option 2 would be awesome as a stand alone model, but harder to couple with a catchment flow model. When I tried to run XBNH in 2D with a river flow input it didn't work because the NH equations try to disperse the river discharge ;).
